src/main/java/org/openstreetmap/atlas/geography/atlas/raw/sectioning/NodeOccurrenceCounter.java

```diff
@@ -0,0 +1,94 @@
+package org.openstreetmap.atlas.geography.atlas.raw.sectioning;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import org.openstreetmap.atlas.geography.Location;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.geography.atlas.raw.slicing.temporary.TemporaryNode;
+
+/**
+ * Keeps track of all {@link TemporaryNode}s created for an atlas {@link Edge} and the occurrence of
+ * it in the underlying polyline.
```

Comment:
Maybe add a comment here saying this is specifically used for way-sectioning

Reply:
Good point!